File: frontend/types/student-project.ts
// Mirrors backend/app/schemas/student_project.py, a read-only adapter
// over `industry_projects` (database/migrations/022_industry_projects.sql).
// Named "student-project" for the type but surfaced at the
// /student/industry-projects route to avoid colliding with the existing
// Student Portfolio "projects" feature.

export interface StudentProjectIndustry {
  id: string;
  company_name: string | null;
  industry_sector: string | null;
  logo_url: string | null;
}

export interface StudentProject {
  id: string;
  title: string;
  description: string;
  location: string | null;
  work_mode: string | null;
  duration_months: number | null;
  team_size: number | null;
  eligibility_criteria: string | null;
  application_deadline: string | null;
  start_date: string | null;
  status: string;
  created_at: string | null;
  industry: StudentProjectIndustry;
  has_applied: boolean;
}

export interface StudentProjectListResponse {
  projects: StudentProject[];
}

File: backend/app/schemas/student_project.py
"""Pydantic schemas for the STUDENT side of Project discovery
(`industry_projects`, database/migrations/022_industry_projects.sql),
read-only. Mirrors app.schemas.student_workshop.
"""

from pydantic import BaseModel


class StudentProjectIndustry(BaseModel):
    id: str
    company_name: str | None = None
    industry_sector: str | None = None
    logo_url: str | None = None


class StudentProject(BaseModel):
    id: str
    title: str
    description: str
    location: str | None = None
    work_mode: str | None = None
    duration_months: int | None = None
    team_size: int | None = None
    eligibility_criteria: str | None = None
    application_deadline: str | None = None
    start_date: str | None = None
    status: str
    created_at: str | None = None
    industry: StudentProjectIndustry
    has_applied: bool


class StudentProjectListResponse(BaseModel):
    projects: list[StudentProject]
